ClientApp/app/(tabs)/create/index.tsx

```diff
@@ -1,25 +1,534 @@
+import themeColors from '@/utils/constants/colors';
+import { hs, vs, mhs } from '@/utils/helpers/uiScaler';
+import { API_ENDPOINTS } from '@/utils/api/endpoints';
+import { createEvent } from '@/services/eventService';
+import { useSelector } from 'react-redux';
+
+const sports = ['Soccer', 'Basketball', 'Tennis', 'Swimming', 'Running', 'Football', 'Rugby'];
+
 const Create = () => {
+  const { control, handleSubmit, formState: { errors }, setValue, reset } = useForm<EventFormData>({
+    defaultValues: {
+      eventType: 'public',
+    },
+  });
+  const router = useRouter();
+  const [isSportTypeModalVisible, setSportTypeModalVisible] = useState(false);
+  const [eventDate, setEventDate] = useState(new Date());
+  const [showDatePicker, setShowDatePicker] = useState(false);
+  const [isSuccessModalVisible, setSuccessModalVisible] = useState(false);
+
+  const user = useSelector((state: { user: any }) => state.user);
+
+  useEffect(() => {
+    if (!user) {
+      console.log('User not found');
+    }
+  }, [user]);
+
+  const navigateHome = () => {
+    router.replace('/(tabs)/home');
+  };
+
+  interface EventFormData {
+    eventName: string;
+    eventType: string;
+    sportType: string;
+    locationName: string;
+    city: string;
+    province: string;
+    country: string;
+    cutOffTime: string;
+    description: string;
+    maxParticipants: string;
+  }
+
+  const onSubmit = async (data: EventFormData) => {
+    try {
+      const eventRequest = {
+        eventName: data.eventName,
+        eventType: data.eventType,
+        sportType: data.sportType,
+        location: {
+          name: data.locationName,
+          streetNumber: '',
+          streetName: '',
+          city: data.city,
+          province: data.province,
+          country: data.country,
+          postalCode: '',
+          addressLine2: '',
+          phoneNumber: '',
+          latitude: '',
+          longitude: '',
+        },
+        date: eventDate.toISOString().split('T')[0],
+        duration: '',
+        maxParticipants: data.maxParticipants,
+        participants: [],
+        createdBy: user.id,
+        teams: [],
+        cutOffTime: data.cutOffTime,
+        description: data.description,
+        isPrivate: data.eventType === 'private',
+        whiteListedUsers: [],
+        requiredSkillLevel: [],
+      };
+
+      await createEvent(eventRequest);
+
+      reset({
+        eventName: '',
+        eventType: 'public',
+        sportType: '',
+        locationName: '',
+        city: '',
+        province: '',
+        country: '',
+        cutOffTime: '',
+        description: '',
+        maxParticipants: '',
+      });
+      setEventDate(new Date());
+      setSuccessModalVisible(true);
+      setTimeout(() => {
+        setSuccessModalVisible(false);
+        navigateHome();
+      }, 2000);
+    } catch (error) {
+      Alert.alert('Error', 'Error occurred while creating the event');
```

Comment:
We should throw an exception here as well.

Reply:
agreed! done